### PR TITLE
Refactored Clean Architecture

### PR DESCRIPTION
**Pull request by Jai0212**
## Description:
This PR refactors the existing codebase to better align with Clean Architecture principles. The structure has been reviewed and adjusted to ensure proper separation of concerns between **entities**, **use cases**, **repositories**, **infrastructure**, and **controllers**. Initial documentation was added, and a linter has been used to clean up the code. The goal is to improve maintainability, testability, and scalability of the system by adhering to Clean Architecture guidelines.

---

## Changes:
1. **Documentation**:  
   - Enhanced documentation for entities, use cases, repositories, and controllers to better explain their responsibilities within the architecture.
   
2. **Separation of Concerns**:  
   - Refactored the code to ensure each layer (Entities, Use Cases, Repositories, Infrastructure, Controllers) has a clear, singular responsibility. 
   - Removed circular dependencies and ensured higher-level layers are independent of lower-level layers.

3. **Dependency Injection**:  
   - Refined the dependency injection mechanism to ensure better flexibility and testability in the codebase.

4. **Refactor Repositories and Infrastructure**:  
   - Repositories now strictly interact with data sources, with infrastructure concerns abstracted accordingly.
   - Business logic was moved from controllers to use cases to better follow the architecture principles.

5. **Code Quality**:  
   - Ran the linter across the codebase and fixed any remaining code quality issues to align with best practices and style guidelines.

6. **Unit Tests**:  
   - Added or refactored unit tests to ensure proper test coverage for all critical components (entities, use cases, repositories).

7. **Controllers Refactor**:  
   - Controllers now only handle HTTP requests and responses, delegating all business logic to the corresponding use cases.

---

## How to Test:
- Review the code to ensure the separation of concerns is properly implemented across the architecture.
- Run the unit tests to verify functionality.
- Check the documentation for clarity on the purpose and responsibilities of each layer.
- Run the linter to ensure no new issues have been introduced.

### CODE REVIEW

**Review by Navnoorsinghmahal — COMMENTED** [review]
Can you tell what part of the code I need to work on?

File: backend/app/use_cases/DatabaseRepositoryInterface.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+from abc import ABC, abstractmethod
```
Comment (Navnoorsinghmahal) [inline backend/app/use_cases/DatabaseRepositoryInterface.py:1]: Can you explain the changes made to this code?

File: backend/app/use_cases/__init__.py [hunk 2]
```diff
@@ -1,9 +1,9 @@
+from .change_password_interactor import ChangePasswordInteractor
 from .generate import Generate
 from .get_headers import GetHeaders
 from .get_last_login_data import GetLastLoginData
 from .get_values_under_header import GetValuesUnderHeader
-from .upload_data import UploadData
-from .share import Share
 from .login_user_interactor import LoginUserInteractor
 from .register_user_interactor import RegisterUserInteractor
-from .change_password_interactor import ChangePasswordInteractor
```
Comment (Navnoorsinghmahal) [inline backend/app/use_cases/__init__.py:9]: Why you removed this line

**Review by Navnoorsinghmahal — COMMENTED** [review]
Great Work

File: backend/app/use_cases/__init__.py (hunk 2 above)
Comment (Jai0212) [reply backend/app/use_cases/__init__.py:9]: I used pylint to fix import statements and ran the package "isort" that fixes the import statements, so this was done automatically. All tests still pass so shouldn't be a probelm

File: backend/app/use_cases/DatabaseRepositoryInterface.py (hunk 1 above)
Comment (Jai0212) [reply backend/app/use_cases/DatabaseRepositoryInterface.py:1]: I added appropriate docstring to the class and fixed the formatting of the code for pylint errors

**Comment by Jai0212** [comment]
> Can you tell what part of the code I need to work on?

In all your files/functions, add the type and return type parameters. Also add appropriate docstrings and all & fix the basic pylint errors

**Pull request merged**